articles/200_migration_guide_from_ros1.md

```diff
@@ -162,3 +162,483 @@ NOTE: to be written
+~~~
+
+The creation of the outgoing message is different in both the namespace and the
+fact that we go ahead and create a shared pointer:
+
+~~~
+//  std_msgs::String msg;
+  auto msg = std::make_shared<std_msgs::msg::String>();
+~~~
+
+In place of `ros::ok()`, we call `rclcpp::ok()`:
+
+~~~
+//  while (ros::ok())
+  while (rclcpp::ok())
+~~~
+
+Inside the publishing loop, we use the `->` operator to access the `data` field
+(because now `msg` is a shared pointer):
+
+~~~
+//    msg.data = ss.str();
+    msg->data = ss.str();
+~~~
+
+To print a console message, instead of using `ROS_INFO()`, we use `printf()`
+(this is temporary, because we don't yet have an equivalent of the `rosconsole`
+package):
+
+~~~
+//    ROS_INFO("%s", msg.data.c_str());
+    printf("%s\n", msg->data.c_str());
+~~~
+
+Publishing the message is very similar, the only noticeable difference being
+that the publisher is now a shared pointer:
+
+~~~
+//    chatter_pub.publish(msg);
+    chatter_pub->publish(msg);
+~~~
+
+Spinning (i.e., letting the communications system process any pending
+incoming/outgoing messages) is different in that the call now takes the node as
+an argument:
+
+~~~
+//    ros::spinOnce();
+    rclcpp::spin_some(node);
+~~~
+
+Sleeping using the rate object is unchanged.
+
+Putting it all together, the new `talker.cpp` looks like this:
+
+~~~
+//#include "ros/ros.h"
+#include "rclcpp/rclcpp.hpp"
+//#include "std_msgs/String.h"
+#include "std_msgs/msg/string.hpp"
+#include <sstream>
+int main(int argc, char **argv)
+{
+//  ros::init(argc, argv, "talker");
+//  ros::NodeHandle n;
+  rclcpp::init(argc, argv);
+  auto node = rclcpp::node::Node::make_shared("talker");
+//  ros::Publisher chatter_pub = n.advertise<std_msgs::String>("chatter", 1000);
+//  ros::Rate loop_rate(10);
+  auto chatter_pub = node->create_publisher<std_msgs::msg::String>("chatter", rmw_qos_profile_default);
+  rclcpp::rate::Rate loop_rate(10);
+  int count = 0;
+//  std_msgs::String msg;
+  auto msg = std::make_shared<std_msgs::msg::String>();
+//  while (ros::ok())
+  while (rclcpp::ok())
+  {
+    std::stringstream ss;
+    ss << "hello world " << count++;
+//    msg.data = ss.str();
+    msg->data = ss.str();
+//    ROS_INFO("%s", msg.data.c_str());
+    printf("%s\n", msg->data.c_str());
+//    chatter_pub.publish(msg);
+    chatter_pub->publish(msg);
+//    ros::spinOnce();
+    rclcpp::spin_some(node);
+    loop_rate.sleep();
+  }
+  return 0;
+}
+~~~
+
+#### Changing the `package.xml`
+
+Starting with ROS 2, only version 2 of the `package.xml` format is supported
+(this format is also supported in ROS 1, but isn't used by all packages).  We
+start by specifying the format version in the `package` tag:
+
+~~~
+<!-- <package> -->
+<package format="2">
+~~~
+
+ROS 2 uses a newer version of `catkin`, called `ament`, which we specify in the
+`buildtool_depend` tag (more specifically, we specify that we want to use
+`ament` to build our package using CMake):
```

Comment:
I removed the parenthetical phrase.  I'd rather not explain what's actually going on under the hood, because almost nobody should need to understand it.